src/schema/nmdc.yaml

```diff
@@ -2324,17 +2346,32 @@ slots:
   metagenome_annotation_id:
     domain: FunctionalAnnotationAggMember
     range: WorkflowExecutionActivity
+    required: true
+    description: The value for slot id for the analysis activity (WorkflowExecutionActivity) that generated functional annotation
```

Comment:
Proposal:

```diff
-    description: The value for slot id for the analysis activity (WorkflowExecutionActivity) that generated functional annotation
+    description: The id of the WorkflowExecutionActivity that generated this FunctionalAnnotationAggMember
```

Also, I find it confusing that the slot name is `..._annotation_id`, but the description says it's an identifier of a `WorkflowExecutionActivity` (as opposed to some kind of "annotation activity" or "annotation"). My background is in software development, not biology—maybe it would be clear to someone with a background in biology.

Reply:
Well, the range of the slot **is** `WorkflowExecutionActivity`. The description doesn't really need to say that's it's value is an `id`,  but the textual `description` does agree well with the logical definition, specifically the `domain` and `range`. I would either leave this is as-is, or improve it by changing the slot `name`. Unfortunately 'generated_by' is already taken!

Reply:
Schema concepts aside (I'll defer to @turbomam on those), I think there's a word (e.g., "the", "a", "this") missing in:
```diff
- that generated functional annotation
+ that generated this functional annotation
```